File: 06-arrays-sorting/05_container_water.c
#include "../test_runner.h"

int max_area(int* height, int n) {
    return -1;
}

int main() {
    TEST_INT("basic",   max_area(ARR(1,8,6,2,5,4,8,3,7), 9), 49);
    TEST_INT("minimal", max_area(ARR(1,1), 2),                 1);
    TEST_INT("U shape", max_area(ARR(4,3,2,1,4), 5),          16);
    TEST_SUMMARY();
}

File: 06-arrays-sorting/05_container_water.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests



# Container With Most Water (LC #11) -- Medium
# Input:  height: List[int] (bar heights, len >= 2)
# Output: int (max water area between two bars)
# Array of heights. Find two lines that form container holding most water.
# Width = distance between lines. Height = shorter line.
#
# Example:
#   [1,8,6,2,5,4,8,3,7] -> 49  (lines at index 1,8: min(8,7)*7=49)
#   [1,1] -> 1
#
#   Key: Two pointers from both ends. Area = min(h[l],h[r]) * (r-l).
#        Move the shorter pointer inward (it limits the area).

def max_area(height):
    pass


run_tests(max_area, [
    (([1,8,6,2,5,4,8,3,7],),  49),
    (([1,1],),                  1),
    (([4,3,2,1,4],),           16),    # 4*4
    (([1,2,1],),                2),    # min(1,1)*2
    (([2,3,4,5,18,17,6],),    17),     # 17*1? no. check: min(2,6)*5=10, min(3,17)*4=12...
    (([1,8,6,2,5,4,8,3,7],),  49),
])
